main_test.go

```diff
@@ -284,8 +284,13 @@ func Test_Main(t *testing.T) {
 		flags = &gatewayFlags{}
 		flagParse = flag.Parse
 	}()
+
 	flags.configFileName = "__INVALID_FILENAME__"
 	main()
+
+	// check with the version flag set
+	flags.version = true
+	main()
```

Comment:
So while this certainly satisfy code coverage, it doesn't assert anything and therefore not a very good test. Please assert that the output you expect actually occurred.

Reply:
I added an assert to ensure the version flag toggles from true to false. This happens when we print the version on the console and return